apps/api/src/app/events/e2e/test-email.e2e.ts

```diff
@@ -0,0 +1,105 @@
+    contentType: 'customHtml',
+    payload: {},
+    subject: 'subject',
+    preheader: 'preheader',
+    content: '<html><head></head><body>Hello world!</body></html>',
+    to: 'to-reply@novu.co',
+  };
+
+  let session: UserSession;
+  let integrationRepository: IntegrationRepository;
+
+  beforeEach(async () => {
+    session = new UserSession();
+    await session.initialize();
+    integrationRepository = new IntegrationRepository();
+  });
+
+  const sendTestEmail = async (body: TestSendEmailRequestDto) => {
+    return await axiosInstance.post(`${session.serverUrl}/v1/events/test/email`, body, {
+      headers: {
+        authorization: session.token,
+      },
+    });
+  };
+
+  const deleteEmailIntegration = async () => {
+    const emailIntegration = await integrationRepository.findOne({
+      channel: ChannelTypeEnum.EMAIL,
+      _environmentId: session.environment._id,
+    });
+    await integrationRepository.delete({ _id: emailIntegration._id, _environmentId: session.environment._id });
+  };
+
+  const deactivateEmailIntegration = async () => {
+    const emailIntegration = await integrationRepository.findOne({
+      channel: ChannelTypeEnum.EMAIL,
+      _environmentId: session.environment._id,
+    });
+    await integrationRepository.update(
+      {
+        _id: emailIntegration._id,
+        _environmentId: session.environment._id,
+      },
+      { active: false }
+    );
+  };
+
+  const reachNovuProviderLimit = async () => {
+    const MAX_NOVU_INTEGRATION_MAIL_REQUESTS = parseInt(process.env.MAX_NOVU_INTEGRATION_MAIL_REQUESTS || '300', 10);
+    const messageRepository = new MessageRepository();
+    for (let i = 0; i < MAX_NOVU_INTEGRATION_MAIL_REQUESTS; i++) {
+      await messageRepository.create({
+        _organizationId: session.organization._id,
+        _environmentId: session.environment._id,
+        providerId: EmailProviderIdEnum.Novu,
+        channel: ChannelTypeEnum.EMAIL,
+      });
+    }
+  };
+
+  it('should allow sending test email with email provider', async function () {
+    const response = await sendTestEmail(requestDto);
+
+    expect(response.status).to.equal(201);
+  });
+
+  it('should allow sending test email with Novu provider', async function () {
+    await deleteEmailIntegration();
+
+    const response = await sendTestEmail(requestDto);
+
+    expect(response.status).to.equal(201);
+  });
+
+  it('should send test email fallbacking to Novu provider when there is no active integration', async function () {
+    await deactivateEmailIntegration();
+
+    const response = await sendTestEmail(requestDto);
+
+    expect(response.status).to.equal(201);
+  });
+
+  it('should not allow sending test email when Novu provider limit is reached', async function () {
+    await deleteEmailIntegration();
+    await reachNovuProviderLimit();
+
+    try {
+      await sendTestEmail(requestDto);
+    } catch (e) {
+      expect(e.response.status).to.equal(400);
```

Comment:
I don't think that 400 Bad Request is the right code when hitting the limit. Find either 409 Conflict or 422 Unprocessable Entity more adequate, as any of them indicate that the request is valid and understood by the server but resource status doesn't allow to proceed. Rather than confuse the client stating the request in wrongly formatted somehow.
Anyway, this is inherited from existing functionality but might be worth to improve it. Feel free to ignore it for this PR and we can address it in a different one.

Reply:
I do agree, I'll update it ;)